### ISSUE TITLE
Multiline codeblocks seen as "superscript" instead of "code" block in confluence

### ISSUE THREAD

**Issue by nado** [szn/confluence.md#12]
Pretty much what the title says, I upload a README file with the following content being rendered incorrectly. Inline `code` renders fine.

```
` ` `ps1
git init
echo * > .gitignore
git remote add origin git@server
git fetch
git checkout origin/master -b master
` ` `
```

**Comment by szn** [comment]
There is a temporary hack to bypass the problem but I am not happy with the results.

Remember that only bare code blocks work (without language specification in the first line).

```
$ pip install --upgrade confluence.md
```

**Comment by mxmo0rhuhn** [comment]
@szn A CI job that runs on an unchanged README.md file of me fails now (since using the new version): 
> requests.exceptions.HTTPError: com.atlassian.confluence.api.service.exceptions.BadRequestException: Content body cannot be converted to new editor format

Is there any way that I can extract the HTML for debugging?

**Comment by szn** [comment]
Sad to hear that @mxmo0rhuhn.

Please install a dev version to see a raw HTML log:

```sh
$ pip install --upgrade --index-url https://test.pypi.org/simple/ confluence.md==0.4.7.dev1
```

And run a command with `-v` option.

**Comment by mxmo0rhuhn** [comment]
Hello, 
I was able to reproduce the issue with the code block.

# Working

a file like this:

```markdown
# Hello world
```

resulting in a page like this:

```html
    <p><strong>Automatic content</strong> - Do not edit this page in Confluence.
    Page automatically generated from: <code>test.md</code></p><hr />
    <h1>Hello world</h1>
```

# Not working

a file like this:

````markdown
    # Hello world

    a file like this should be converted to a confluence page

    ```markdown
    # Hello world
    ```
````

resulting in a page like this:

```html
    <p><strong>Automatic content</strong> - Do not edit this page in Confluence.
    Page automatically generated from: <code>test_code.md</code></p><hr />
    <h1>Hello world</h1>

<p>a file like this should be converted to a confluence page</p>

<div class="codehilite">
<pre><span></span><code><span class="gh"># Hello world</span>
</code>
</div>
```

which is subsequently not accepted by the confluence API:
```
requests.exceptions.HTTPError: com.atlassian.confluence.api.service.exceptions.BadRequestException: Content body cannot be converted to new editor format
```

At first glance, it looks to me like there is a `<pre>` tag generated that is not used and not closed.

**Comment by szn** [comment]
Thanks @mxmo0rhuhn 

I am not proud of this fix but it seems to work:

```
pip install --upgrade --index-url https://test.pypi.org/simple/ confluence.md==0.4.7.dev2
```

I believe I reached the limits of `markdown2` package. For code blocks to render properly I would need to either replace the md2html translator or perform more intense HTML post-processing. Marked as TODO.

**Comment by mxmo0rhuhn** [comment]
Thank you for the super fast fix! There is a special case that does not work now: 

````markdown
```bash
echo "hello, world!"
```

```python
print("hello, world!")
```
````

resulting in a page like this:

```html
    <p><strong>Automatic content</strong> - Do not edit this page in Confluence.
    Page automatically generated from: <code>test_code.md</code></p><hr />
    <div class="codehilite">
<code><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Hello, World!&quot;</span>
</code>
</div>

<div class="codehilite">
<pre><span></span><code><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Hello, World!&quot;</span><span class="p">)</span>
</code>
</div>
```

from a first look it seems like in this setup the regex is not able to match the opening `<pre>` tag. interestingly, if there is only one code block in the file, the `<pre>` tag is matched correctly and the upload works as expected.

**Comment by mxmo0rhuhn** [comment]
Yay it works! All files that I tested with were correctly uploaded to Confluence 👍 

Thanks a lot for your continued effort!

**Comment by szn** [comment]
Thanks for testing!

**Comment by szn** [comment]
@mxmo0rhuhn `0.4.7.dev3` was published as `0.4.8` (except debug html log). You can:

```sh
$ pip install --upgrade confluence.md
```
